### PR TITLE
Cleanup `pkg/storage`

### PR DESCRIPTION
**Pull request by bkcsoft**
- Split it storage.proto into logical units
- `make` target for generating protobuf files
- Use `google.protobuf.Empty` instead of defining our own
- storage.storage => storage.LocalStorage, and return on New

### CODE REVIEW

**Comment by bkcsoft** [comment]
The reason for the renaming comes mainly from [googles naming conventions](https://cloud.google.com/apis/design/naming_convention#method_names), but I'm not following it religiously :wink: For example `Repository.SetDescription` is neither a `Create` (doesn't create anything) or a `Update` (not an update in the traditional sense).

**Comment by bkcsoft** [comment]
Aaaaand done... for now :smile: 

/cc @metalmatze

File: pkg/storage/storage_test.go [hunk 1]
```diff
@@ -8,6 +8,9 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestInterface(t *testing.T) {
+	assert.Implements(t, (*Storage)(nil), &LocalStorage{})
```
Comment (metalmatze) [inline pkg/storage/storage_test.go]: This should be tested here:
https://github.com/sourcepods/sourcepods/blob/31766d2c90d98e962b5ac171c9912f2cb65296e0/cmd/storage/storage.go#L172

At this point the LocalStorage needs to implement that interface in code. Not entirely sure when there are multiple implementations, or if it fails at runtime only?!

We'll keep this anyway. More a comment on this test in general.

File: pkg/storage/client.go [hunk 2]
```diff
@@ -81,14 +90,15 @@ func (c *Client) Branches(ctx context.Context, owner string, name string) ([]Bra
...
+// Commit returns a single commit from a given repository
+func (c *Client) Commit(ctx context.Context, owner, name, rev string) (Commit, error) {
 	req := &CommitRequest{
 		Owner: owner,
 		Name:  name,
 		Rev:   rev,
 	}
 
-	res, err := c.client.Commit(ctx, req)
+	res, err := c.commits.Create(ctx, req)
```
[the remark this answers is not in the document]
Comment (bkcsoft) [reply pkg/storage/client.go]: Supposed to be `commits.Get`, but yeah, fixed :slightly_smiling_face:

File: pkg/storage/storage_test.go (hunk 1 above)
Comment (bkcsoft) [reply pkg/storage/storage_test.go]: I usually only test the package I'm in, only doing `make test` before commiting. Hence this test.

But true, that line will essentially test the assertion on `go build` :)

**Pull request merged**